kernelci/kbuild.py

```diff
@@ -699,16 +699,17 @@ def submit(self, retcode, dry_run=False):
             },
             'child_nodes': [],
         }
-        # TODO: Fix this hack
+        # Update node from API, as we might have new fields
+        # such as k8s_context
+        nodeid = self._node['id']
+        self._node = api.node.get(nodeid)
```

Comment:
Can this be done at the start of the function? In #2342 the node data with a new field in upload_artifacts(), IIUC this change would be lost if the node is refreshed afterwards.